app/Jobs/ProvisionQueryserviceNamespaceJob.php

```diff
@@ -75,31 +81,33 @@ public function handle()
             ],
         ]);
 
-        $response = curl_exec($curl);
-        $err = curl_error($curl);
-
-        curl_close($curl);
+        $response = $this->request->execute(); 
+        $err = $this->request->error();
 
+        $this->request->close();
         if ($err) {
-            $this->fail(
-                new \RuntimeException('cURL Error #:'.$err)
-            );
-
-            return; //safegaurd
+            $this->fail( new \RuntimeException('cURL Error #:'.$err) );
+            return;
         } else {
             if ($response === 'CREATED: '.$this->namespace) {
                 $qsns = QueryserviceNamespace::create([
                     'namespace' => $this->namespace,
-                    //'internalHost' => $this->internalHost,
+                    //'internalHost' => $this->internalHost, TODO is this required?
                     'backend' => $queryServiceHost,
                 ]);
             // TODO error if $qsns is not actually created...
+            } else if( $response === 'EXISTS: ' .$this->namespace ) {
```

Comment:
I don't think we want to have this as a "good case".
If the NS already exists then that hints that we have a conflict somehow.

As the namespace comes from a random string this could indiacte we havn't been as random as we expect.
https://github.com/wbstack/api/pull/155/files#diff-d92a008b006bf2d371a959dee8f8d3be5724b5fce1655f46102404cbe4f27977R33

If this were to happen, but we count this as acceptable, wikis could end up sharing a namespace?

Reply:
Ok, 

What do you think we should do, fail the job and log an error?

Reply:
I made it fail and log an error now

Reply:
Perfect!